lib/Dialect/FIRRTL/Transforms/Lint.cpp

```diff
@@ -0,0 +1,58 @@
+//===- Lint.cpp -------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/IR/Threading.h"
+#include "llvm/ADT/APSInt.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct LintPass : public LintBase<LintPass> {
+  void runOnOperation() override {
+    auto fModule = getOperation();
+    (void)failableParallelForEach(
+        fModule.getContext(), fModule.getOps(),
+        [&](Operation &op) { return checkAssert(&op); });
+  };
+  LogicalResult checkAssert(Operation *op) {
+    Value predicate;
+    if (auto a = dyn_cast<AssertOp>(op)) {
+      if (auto constant = a.getEnable().getDefiningOp<firrtl::ConstantOp>())
+        if (constant.getValue().isOne()) {
+          predicate = a.getPredicate();
+        }
+    } else if (auto a = dyn_cast<VerifAssertIntrinsicOp>(op))
+      predicate = a.getProperty();
+
+    if (!predicate)
+      return success();
+    if (auto constant = predicate.getDefiningOp<firrtl::ConstantOp>())
+      if (constant.getValue().isZero())
+        return op->emitOpError(
+            "is guaranteed to fail simulation, as the predicate is "
+            "constant false");
```

Comment:
Maybe attach a note for the location of the constant op like for the reset case?

Reply:
Good idea, added them.